test/Microsoft.Extensions.DiagnosticAdapter.Test/DiagnosticSourceAdapterTest.cs

```diff
@@ -79,6 +90,19 @@ public void IsEnabled_FalseForNonenlistedEvent()
         }
 
         [Fact]
+        public void CallingWriteWithWrongTypeThrows()
```

Comment:
While we're on it, if I do
```
[DiagnosticName("Bar")]
public void OnBar(int baz)
{
    ...
}
...
adapter.Write("Bar", new { baz = (int?)null });
```
We would expect that to throw right? Because the actual behavior is that it calls OnBar with baz=0.